### PR TITLE
Add Artist Info Pages

### PR DESCRIPTION
**Pull request by amCap1712**
Add Artist Info pages with following specs:-
1) Bio page to display details of country, birth, gender and Wikipedia Bio Summary if present.
2) Releases of the artist with cover art.
3) URL relationships of the artist.

### CODE REVIEW

**Comment by Miguel-Herrero** [comment]
@amCap1712 Good job. I suggest that while you are refactoring the app you be as “verbose” as you can and use comments. It is a good way of communicating your intentions and let the people help you to improve code if something can be done in a better way 👍

File: app/src/main/java/org/metabrainz/mobile/activity/ArtistActivity.java [hunk 1]
```diff
@@ -25,66 +20,38 @@
...
-    @Override
-    protected void onResume() {
-        super.onResume();
-        String id = getIntent().getStringExtra(Extra.ARTIST_MBID);
-        if(id != null && !id.isEmpty()) artistViewModel.setMBID(id);
-            mbid = id;
-        artistViewModel.setMBID(mbid);
-        artistViewModel.getArtistData().observe(this, this::setArtist);
+        // Whenever the artist changes, redraw the information
+        artistViewModel.initializeArtistData().observe(this, this::setArtist);
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply app/src/main/java/org/metabrainz/mobile/activity/ArtistActivity.java]: Subscribe to the empty live data and then ask the view model to update artist live data. The approach has the benefit of eliminating extra calls to update artist info when it is not required. 

Example: The view model is shared between the activity and fragments. The activity and each of the fragments will need to subscribe to the live data independently. In the earlier approach, the artist data was queried whenever getArtistData() was invoked, the repository performed an update on the artist data. This approach led a lot of unneeded network request. A better solution which is currently followed is that there is a separate method to subscribe to live data and another one to update the artist info. The initializeLiveData method acts like a getter method.

[the remark this answers is not in the document]
Comment (Miguel-Herrero) [reply app/src/main/java/org/metabrainz/mobile/activity/ArtistActivity.java]: What I thought is that ViewModel is [shared between Activity and Fragments](https://developer.android.com/topic/libraries/architecture/viewmodel.html#sharing), so once the Activity has requested the info to the repository (via ViewModel), once Fragments subscribe that info will be returned from ViewModel without hitting repository (or, at least, without hitting the API).

I see two boundaries:
- **ViewModel** for Activity & Fragments: if they are requesting the same data, ViewModel returns the already fetched data (shared data) without hitting higher layers.
- **Repository** for ViewModel: if different ViewModels are requesting the same data, Repository returns the “cached” value (shared data)

(In Clean Architecture theory, the ViewModel boundary would be moved to a **UseCase**, as ViewModel would NOT request data to repository on its own)

[the remark this answers is not in the document]
Comment (amCap1712) [reply app/src/main/java/org/metabrainz/mobile/activity/ArtistActivity.java]: The ViewModel is indeed shared between activity and fragments. But, once the fragments subscribe the repository  being hit. So, that's why I made this change.

**Pull request merged**